src/main/java/com/flightstats/hub/app/HubMain.java

```diff
@@ -3,110 +3,203 @@
     private static final DateTime startTime = new DateTime();
 
     public static void main(String[] args) throws Exception {
         if (args.length == 0) {
             throw new UnsupportedOperationException("HubMain requires a property filename, 'useDefault', or 'useEncryptedDefault'");
         }
         HubProperties.loadProperties(args[0]);
-        start();
+        new HubMain().run();
     }
 
-    static void start() throws Exception {
+    public void run() throws Exception {
         Security.setProperty("networkaddress.cache.ttl", "60");
         startZookeeperIfSingle();
-        HubJettyServer server = startServer();
+        Server server = startServer();
 
         final CountDownLatch latch = new CountDownLatch(1);
-        Runtime.getRuntime().addShutdownHook(new Thread() {
-            public void run() {
-                logger.info("Jetty Server shutting down...");
-                latch.countDown();
-            }
-        });
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            log.info("Jetty Server shutting down...");
+            latch.countDown();
+        }));
         latch.await();
-        logger.warn("calling shutdown");
+
+        log.warn("calling shutdown");
         HubProvider.getInstance(ShutdownManager.class).shutdown(true);
-        server.halt();
-        logger.info("Server shutdown complete.  Exiting application.");
+
+        server.stop();
+    }
+
+    private void startZookeeperIfSingle() {
+        new Thread(() -> {
+            String zkConfigFile = HubProperties.getProperty("runSingleZookeeperInternally", "");
+            if ("singleNode".equals(zkConfigFile)) {
+                log.warn("using single node zookeeper");
+                ZookeeperMain.start();
+            }
+        }).start();
     }
 
-    public static HubJettyServer startServer() throws IOException {
-        ResourceConfig resourceConfig = new ResourceConfig();
-        resourceConfig.register(new ObjectMapperResolver(HubBindings.objectMapper()));
-        resourceConfig.register(JacksonJsonProvider.class);
-        resourceConfig.registerClasses(CORSFilter.class,EncodingFilter.class, StreamEncodingFilter.class, GZipEncoder
-                        .class,
-                DeflateEncoder.class);
+    @VisibleForTesting
+    public Server startServer() throws Exception {
+        List<AbstractModule> guiceModules = buildGuiceModules();
+        Injector injector = Guice.createInjector(guiceModules);
 
-        List<Module> modules = new ArrayList<>();
+        HubProvider.setInjector(injector);
+        HubServices.start(HubServices.TYPE.BEFORE_HEALTH_CHECK);
+
+        // build Jetty server
+        Server server = new Server();
+        HttpConfiguration httpConfig = new HttpConfiguration();
+        SslContextFactory sslContextFactory = getSslContextFactory();
+        if (null != sslContextFactory) {
+            httpConfig.addCustomizer(new SecureRequestCustomizer());
```

Comment:
That's how it was before I got in there and I was trying to limit my changes. It looks like new code because I merged the HubMain and HubJettyServer classes.